beetmoverscript/script.py

```diff
@@ -76,10 +113,64 @@
     # the checksums file
     add_release_props_to_artifacts(context, release_props_file)
 
+
+async def push_to_releases(context):
+    pass
+
+
+async def push_to_staging(context):
+    pass
+
+
+action_map = {
+    'push-to-nightly': push_to_nightly,
+    'push-to-candidates': push_to_candidates,
+    'push-to-releases': push_to_releases,
+    'push-to-staging': push_to_staging
+}
+
+
+# async_main {{{1
+async def async_main(context):
+    # determine the task and make a quick validation check against its schema
+    context.task = get_task(context.config)  # e.g. $cfg['work_dir']/task.json
+    validate_task_schema(context)
+
+    # determine the task action to toggle the right behavior
+    context.cert_name = get_task_cert_type(context.task)
+    context.action = get_task_beetmover_action(context.cert_name, context.task)
+
+    if action_map.get(context.action):
+        await action_map[context.action](context)
+    else:
+        log.critical("Unknow action {}!".format(context.action))
+        sys.exit(3)
+
     log.info('Success!')
 
 
-async def move_beets(context, artifacts_to_beetmove, manifest):
+async def move_candidates_beets(context, artifacts_to_beetmove, manifest):
+    beets = []
+    for locale in artifacts_to_beetmove:
+        for artifact in artifacts_to_beetmove[locale]:
+            source = artifacts_to_beetmove[locale][artifact]
+            artifact_pretty_name = manifest['mapping'][locale][artifact]['s3_key']
+            destinations = [os.path.join(manifest["s3_prefix_candidates"],
```

Comment:
@escapewindow If you think the `candidates` templates approach (see question below in the template) is better than what we currently have for nightlies (as s3_key + latest and dated prefixes) we could tweak the `nightly` templates as well and reuse the same logic, hence getting rid of this additional `move_candidatess_beets`. The same with the `push-to-nightly` vs `push-to-candidates` which currently only differ by this function call.